google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -822,11 +832,49 @@ private void put(SpannerException e) {
     }
 
     private PooledSession take() throws SpannerException {
-      SessionOrError s = Uninterruptibles.takeUninterruptibly(waiter);
-      if (s.e != null) {
-        throw newSpannerException(s.e);
+      long currentTimeout = options.getInitialWaitForSessionTimeoutMillis();
+      while (true) {
+        Tracer tracer = Tracing.getTracer();
+        Scope waitScope = tracer.spanBuilder(WAIT_FOR_SESSION).startScopedSpan();
```

Comment:
The standard way to create a scoped span is in a try block and then you don't explicitly have to close the scope [[example](https://opencensus.io/quickstart/java/tracing/#create-a-span)].

Reply:
Changed to try-with-resources block.